File: java/L000/LeetCode0066.java
package L000;

public class LeetCode0066 {
    public int[] plusOne(int[] digits) {
        int C = 1;
        for (int i = digits.length - 1;i >= 0;--i) {
            int t = digits[i] + C;
            digits[i] = t % 10;
            C = t / 10;
        }
        if (C == 0) {
            return digits;
        } else {
            int[] res = new int[digits.length + 1];
            res[0] = C;
            for (int i = 0;i < digits.length;++i) {
                res[i + 1] = digits[i];
            }
            return res;
        }
    }
}

File: cpp/L000/LeetCode0066.cc
#include <algorithm>
#include <vector>
using std::vector;

class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        vector<int> res;
        int carry = 1;
        for (auto it{digits.rbegin()};it != digits.rend();++it) {
            int x = *it + carry;
            res.emplace_back(x % 10);
            carry = x / 10;
        }
        if (carry > 0) {
            res.emplace_back(carry);
        }
        std::reverse(res.begin(), res.end());
        return res;
    }
};
